packages/core/src/semantic-icon-provider/SemanticIconProvider.tsx

```diff
@@ -0,0 +1,76 @@
+import {
+  CalendarIcon,
+  ChevronDownIcon,
+  ChevronLeftIcon,
+  ChevronRightIcon,
+  ChevronUpIcon,
+  CloseIcon,
+  ErrorSolidIcon,
+  InfoSolidIcon,
+  OverflowMenuIcon,
+  SuccessTickIcon,
+  TearOutIcon,
+  UserSolidIcon,
+  WarningSolidIcon,
+} from "@salt-ds/icons";
+import type React from "react";
+import { type FC, type ReactNode, createContext, useContext } from "react";
+
+type IconMap = {
```

Comment:
`Semantic*` prefix feels a little bit redundant ..? maybe cross reference the names with Paul's filie?

Maybe one of these types can be exported. Would be useful, as a user to do, something like 
```
const myMap = { WarningIcon: MyWarningIcon } satisfies SemanticIconMap; 
```

Reply:
done